File: 백준/Silver/16953. A → B/A → B.cc
#include <iostream>
#define endl "\n"
using namespace std;

int main()
{
    long long A, B;
    cin >> A >> B;

    int cnt = 0;
    bool flag = true;
    while (A != B)
    {
        if (B % 10 == 1)
        {
            B /= 10;
            cnt++;
        }

        else if (B % 2 == 0 && B >= A)
        {
            B /= 2;
            cnt++;
        }

        else
        {
            flag = false;
            break;
        }
    }

    if (!flag)
    {
        cout << -1;
    }
    else
    {
        cout << cnt + 1;
    }
}

File: 백준/Silver/16953. A → B/A → B.py
A, B = map(int, input().split())

cnt = 1
while B > A:
    if B % 2 == 0:  # 짝수
        B //= 2
    elif B % 10 == 1:  # 마지막 자리 1
        B //= 10
    else:  # 짝수도 아니고 마지막 자리 1도 아님 or 연산 종료
        break
    cnt += 1

if B == A:
    print(cnt)
else:
    print(-1)
